include/mbedtls/ssl.h

```diff
@@ -801,6 +801,24 @@ typedef struct mbedtls_ssl_key_cert mbedtls_ssl_key_cert;
 typedef struct mbedtls_ssl_flight_item mbedtls_ssl_flight_item;
 #endif
 
+#if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_CLI_C)
+#define MBEDTLS_SSL_EARLY_DATA_STATUS_UNKNOWN           0
+#define MBEDTLS_SSL_EARLY_DATA_STATUS_NOT_SENT          1
+#define MBEDTLS_SSL_EARLY_DATA_STATUS_INDICATION_SENT   2
+#define MBEDTLS_SSL_EARLY_DATA_STATUS_REJECTED          3
+#define MBEDTLS_SSL_EARLY_DATA_STATUS_ACCEPTED          4
+#endif
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
+
+typedef enum
+{
+    MBEDTLS_SSL_TICKET_ALLOW_EARLY_DATA = 1,
+    MBEDTLS_SSL_TICKET_ALLOW_DHE_RESUMPTION = 2,
+    MBEDTLS_SSL_TICKET_ALLOW_PSK_RESUMPTION = 4,
+} mbedtls_ssl_ticket_flags;
```

Comment:
```suggestion
typedef uint8_t mbedtls_ssl_ticket_flags;

#define MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_RESUMPTION              ( 1u << 0 )
#define MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_PSK_EPHEMERAL_RESUMPTION    ( 1u << 2 )
#define MBEDTLS_SSL_TLS1_3_TICKET_HAS_EARLY_DATA_INDACTION          ( 1u << 3 )
```

- `ticket_flags` will be serialized and the size of enum is depends on compiler. So, replace it with `uint8_t`.
- Align the value with `MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_*`
- See https://github.com/Mbed-TLS/mbedtls/issues/6326#issuecomment-1307129452, Until now, I think `allow_early_data` should be renamed

Reply:
Seems we don't directly use mbedtls_ssl_ticket_flags in this PR, should we change it here or in the later PRs once we define variables with it?

Reply:
> 1. mbedtls_ssl_ticket_flags is not directly used in this PR, should we change it here or in the later PRs once we define variables with it?

Rollback to `development` if you think so. 

> 
>     2. Why should we define the macro to 1,4,8? not 1,2,4?

See above. Read comment message firstly before you reply something.

> 
>     3. Why the name MBEDTLS_SSL_TLS1_3_TICKET_HAS_EARLY_DATA_INDACTION doesn't follow up the ALLOW_*?

See above,  Read comment message firstly before you reply something.